sandy/core/cov.py

```diff
@@ -812,6 +810,76 @@ def gls_update(self, S, Vy_extra, threshold=None):
         Vx_post = Vx_prior - A.dot(Vx_prior)
         return self.__class__(pd.DataFrame(Vx_post, index=index, columns=columns))
 
+    def _constrained_ls_sensitivity(self, S, threshold=None):
+        """
+        Method to obtain sensitivity according to constrained Least-Squares:
+        (S*Vx_prior*S.T)^(-1)*S*Vx_prior
+
+        Parameters
+        ----------
+        S : 2D iterable
+            Sensitivity matrix (MXN).
+
+        Returns
+        -------
+        `pd.DataFrame`
+            constrained Least-Squares sensitivity.
+
+        Example
+        -------
+        >>> S = np.array([[1, 2], [3, 4]])
+        >>> sensitivity = CategoryCov.from_var([1, 1])
+        >>> sensitivity._constrained_ls_sensitivity(S)
+                      0	              1
+        0	-2.00000e+00	1.50000e+00
+        1	1.00000e+00	  -5.00000e-01
+        """
+        # Data in a appropiate format
+        Vx_prior = self.data.values
+        Vy_extra = pd.DataFrame(0, index=self.data.index,
+                                columns=self.data.columns)
+        S_ = pd.DataFrame(S)
+        index = S_.index
+        columns = S_.columns
+        # constrained Least Squares sensitivity
+        M = self._gls_M(S_, Vy_extra)
+        M_inv = sandy.CategoryCov(M).invert().data.values
+        sensitivity = M_inv.dot(S_.values).dot(Vx_prior)
+        if threshold is not None:
+            sensitivity[sensitivity < threshold] = 0
+        return pd.DataFrame(sensitivity, index=index, columns=columns)
+
+    def constrained_ls_update(self, S, threshold=None):

```

Comment:
this looks like the gls_update but without Vy_extra in the inverse matrix

Reply:
can we call it `_constrained_gls_update` for consistency?

also, add a note and say that this method is equivalent to `_gls_update` but for a constrained system.

Reply:
Is the gls_update without Vy_extra. For consistency i call constrained_gls_update (whithout the private)

Reply:
yes, sorry...the private part was my mistake